modules/study/src/main/BSONHandlers.scala

```diff
@@ -137,81 +139,81 @@ object BSONHandlers:
       comments       = doc.getAsOpt[Comments](F.comments) getOrElse Comments.empty
       gamebook       = doc.getAsOpt[Gamebook](F.gamebook)
       glyphs         = doc.getAsOpt[Glyphs](F.glyphs) getOrElse Glyphs.empty
-      score          = doc.getAsOpt[Score](F.score)
+      eval           = doc.getAsOpt[Score](F.score).map(_.eval)
       clock          = doc.getAsOpt[Centis](F.clock)
-      crazy          = doc.getAsOpt[Crazyhouse.Data](F.crazy)
+      crazyData      = doc.getAsOpt[Crazyhouse.Data](F.crazy)
       forceVariation = ~doc.getAsOpt[Boolean](F.forceVariation)
-    } yield Node(
-      id,
-      ply,
-      Uci.WithSan(uci, san),
-      fen,
-      check,
-      shapes,
-      comments,
-      gamebook,
-      glyphs,
-      score,
-      clock,
-      crazy,
-      Node.emptyChildren,
-      forceVariation
+    } yield Branch(
+      id = id,
+      ply = ply,
+      move = Uci.WithSan(uci, san),
+      fen = fen,
+      check = check,
+      shapes = shapes,
+      comments = comments,
+      gamebook = gamebook,
+      glyphs = glyphs,
+      eval = eval,
+      clock = clock,
+      crazyData = crazyData,
+      children = Branches.empty,
+      // TODO we probably need to have access to the variant to set the opening correctly
```

Comment:
Currently the `opening` field is only set when sending to the client, but this is not type-safe. A better way would be to set it right when reading from the DB but we need to have access to `variant`, which not stored in the document, meaning we cannot use default BSON readers etc, for `Branch`, `Root` and eventually `Chapter`, which makes for a bunch of additional manual code like that: 

```scala
def readDoc(doc: Doc): Chapter

...

meta = ....
root = readRoot(meta.variant)
Chapter (
... fields
)
```

To save some gain, I've thought about using macro generated code for `Writing` but not for reading but could not make it work, and also `BSONWriter` is not implemented anywhere in Lila directly so this is probably not a good approach.